### PR TITLE
GCC 13 support

### PR DESCRIPTION
**Pull request by chausner**
This PR continues where https://github.com/David-Haim/concurrencpp/pull/51 left off and makes the library compatible with GCC 13 while not including the workarounds needed for GCC 11/12. The TSAN test suite passes successfully for me when building with GCC 13.

There are two warnings that GCC 13 triggers, one related to the use of `std::hardware_destructive_interference_size`, the other about freeing a non-heap object in `result_state::delete_self`. I have added pragmas to suppress both warnings but if you prefer to keep them, I can revert the changes.

The PR adds GCC 13 builds and tests to the CI for Ubuntu 22.04 and macOS 12. Some of the timer-related tests fail on macOS, as it has been the case for Clang. Also, I had to remove the TSAN variant from the build matrix on macOS because linking failed due to missing TSAN libraries.

### CODE REVIEW

File: include/concurrencpp/threads/cache_line.h [hunk 1]
```diff
@@ -6,9 +6,16 @@
...
+#        pragma GCC diagnostic ignored "-Winterference-size"
+#    endif
+constexpr inline std::size_t CRCPP_CACHE_LINE_ALIGNMENT = std::hardware_destructive_interference_size;
+#    if defined(CRCPP_GCC_COMPILER) 
+#        pragma GCC diagnostic pop
+#    endif
 #else
-#    define CRCPP_CACHE_LINE_ALIGNMENT 64
+constexpr inline std::size_t CRCPP_CACHE_LINE_ALIGNMENT = 64;
 #endif
```
Comment (chausner) [inline include/concurrencpp/threads/cache_line.h:19]: I turned this preprocessor definition into a `constexpr inline` in order to be able to wrap it in `#pragma GCC diagnostic push/pop`.

**Comment by chausner** [comment]
I added CI builds for gcc 13 on Ubuntu and macOS, see the updated PR description.

**Comment by David-Haim** [comment]
Amazing!
Im vacationing, I will look at it next week.
Im so glad we will finally have a GCC support!

**Comment by David-Haim** [comment]
I'm approving this.
thanks @chausner for the patience and persistence! concurrencpp will HUGELY benefit from your contribution!

**Comment by tocic** [comment]
Great news! @David-Haim do you plan to release a new version soon? I want to package the library from the github release.

**Comment by chausner** [comment]
> I'm approving this. thanks @chausner for the patience and persistence! concurrencpp will HUGELY benefit from your contribution!

You're welcome!

**Pull request merged**